instrumented/integration/src/androidTest/kotlin/com/datadog/android/sdk/integration/rum/KioskTrackingTest.kt

```diff
@@ -0,0 +1,139 @@
+
+        ConditionWatcher {
+            verifyExpectedEvents(mockServerRule.getRequests(), expectedEvents)
+            true
+        }.doWait(timeoutMs = FINAL_WAIT_MS)
+    }
+
+    override fun runInstrumentationScenario(
+        mockServerRule: RumMockServerActivityTestRule<KioskSplashPlaygroundActivity>
+    ): List<ExpectedEvent> {
+        val activity = mockServerRule.activity
+        val instrumentation = InstrumentationRegistry.getInstrumentation()
+        val expectedEvents = mutableListOf<ExpectedEvent>()
+        val firstViewUrl = activity.javaClass.canonicalName!!.replace(
+            '.',
+            '/'
+        )
+        val secondViewUrl = KioskTrackedPlaygroundActivity::class.java.canonicalName!!.replace(
+            '.',
+            '/'
+        )
+
+        instrumentation.waitForIdleSync()
+        waitForPendingRUMEvents()
+
+        expectedEvents.add(ExpectedApplicationStartActionEvent())
+        expectedEvents.add(
+            ExpectedApplicationLaunchViewEvent(
+                docVersion = 2
+            )
+        )
+
+        // Stop launch view
+        expectedEvents.add(
+            ExpectedApplicationLaunchViewEvent(
+                docVersion = 3
+            )
+        )
+
+        // No events on this view - one for view stop view / stop session
+        expectedEvents.add(
+            ExpectedViewEvent(
+                firstViewUrl,
+                docVersion = 2,
+                viewArguments = mapOf(),
+                sessionIsActive = false
+            )
+        )
+
+        Thread.sleep(2000)
+        onView(withId(R.id.end_session)).perform(click())
+        instrumentation.waitForIdleSync()
+        Thread.sleep(500)
+
+        onView(withId(R.id.start_kiosk)).perform(click())
+        instrumentation.waitForIdleSync()
+
+        // one for view start
+        expectedEvents.add(
+            ExpectedViewEvent(
+                secondViewUrl,
+                docVersion = 2,
+                viewArguments = mapOf(),
+                sessionIsActive = true
+            )
+        )
+
+        // one for view stop
+        expectedEvents.add(
+            ExpectedViewEvent(
+                secondViewUrl,
+                docVersion = 3,
+                viewArguments = mapOf(),
+                sessionIsActive = true
+            )
+        )
+
+        onView(withId(R.id.kiosk_button)).perform(click())
+        instrumentation.waitForIdleSync()
+        onView(withId(R.id.kiosk_back_button)).perform(click())
+        instrumentation.waitForIdleSync()
+
+        // No events on this view - one for view stop view / stop session
+        expectedEvents.add(
+            ExpectedViewEvent(
+                firstViewUrl,
+                docVersion = 2,
+                viewArguments = mapOf(),
+                sessionIsActive = false
+            )
+        )
+
+        onView(withId(R.id.end_session)).perform(click())
```

Comment:
why do we expect `sessionIsActive = false` before we click on `End Session`? shouldn't `onView(withId(R.id.end_session)).perform(click())` be above?

Reply:
This is a weird thing about how these tests are written. I need to capture current scope before I actually end the session, so I have to create the expected event before I click the button. The same reason the other tests add the expected "stop view" before actually stopping the view.